### DISCUSSION TITLE
[REQUEST] Support for dockable probe

### DISCUSSION THREAD

**Discussion by CarnageMarkus** [jschuh/klipper-macros#167] [Ideas]
Hi, first of all thanks for great set of macros.
I successfully setup dockable probe with this library, but it was rather problematic (lots of strange errors).

To integrate dockable probe, I need to wrap original `BED_MESH_CALIBRATE ` to do docking operations before and after calling the original `BED_MESH_CALIBRATE`. Since this library renames the original `BED_MESH_CALIBRATE ` macro, this requires edits to ensure the library calls my custom calibrate function with docking operations, and then that it actually calls the `_KM_BED_MESH_CALIBRATE_BASE`

I think this could be made easier if we could customize the name of a command that `bed_mesh_calibrate_fast` calls
Similarly, the issue with recursive G28 could be also resolved by having option to customize macro called instead of G28

**Comment by jschuh** [comment]
Klipper already provides wrapping as a generic mechanism for that sort of customization, and I don't want to complicate the macros by adding another layer of indirection for doing the same thing.

**Comment by HitLuca** [comment]
Having the same issue, @CarnageMarkus how did you fix it on your side?

**Comment by wreckedemm** [comment]
@HitLuca I'm not @CarnageMarkus but i fixed mine by using this as my start Gcode. I am also using the klicky macros at https://github.com/Harrypulvirenti/Klack-Probe-Macros

M190 S0 ; Remove this if autoemit_temperature_commands is off in Prusa Slicer 2.6 and later
M109 S0 ; Remove this if autoemit_temperature_commands is off in Prusa Slicer 2.6 and later
_PRINT_START_PHASE_INIT EXTRUDER={first_layer_temperature[initial_tool]} BED=[first_layer_bed_temperature] MESH_MIN={first_layer_print_min[0]},{first_layer_print_min[1]} MESH_MAX={first_layer_print_max[0]},{first_layer_print_max[1]} LAYERS={total_layer_count} NOZZLE_SIZE={nozzle_diameter[0]}
G28 X Y 
_PRINT_START_PHASE_PREHEAT
Attach_Probe_Lock
_PRINT_START_PHASE_PROBING
Dock_Probe_Unlock
_PRINT_START_PHASE_EXTRUDER
; Insert custom gcode here.
_PRINT_START_PHASE_PURGE

I do not know if this is the most efficient way, but it is working.

**Reply by HitLuca** [reply]
After some digging I ended up having to only update one line in `bed_mesh_fast.cfg`: at the end of the `bed_mesh_calibrate_fast` macro, there's a call to `{calibrate_cmd}{%for k in params%}{' '~k~'="'~params[k]~'"'}{%endfor%}` which I replaced with `BED_MESH_CALIBRATE{%for k in params%}{' '~k~'="'~params[k]~'"'}{%endfor%}`. Full and adaptive mesh probing works without issues, but I am not sure if there are implications which I haven't encountered yet

**Reply by wreckedemm** [reply]
Thanks for the info. If something breaks i'll have to try this. haha. Things are working on my end and i dont want to mess with it
